### PR TITLE
`TabView` updates, WUX `IconSource` support

### LINKED ISSUE #4908: Integrate latest TabView changes
## What would you like to be added:

Integrate latest `TabView` updates to keep in sync with WinUI.

## Why is this needed:

WinUI parity

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### LINKED ISSUE #6167: WUX IconSources are not implemented
## Current behavior

`IconSource` types are implemented in MUX but not in WUX.

## Expected behavior

Should be implemented in both.

## How to reproduce it (as minimally and precisely as possible)

-

## Workaround

N/A

## Environment

Nuget Package:

- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other:

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #4908, closes #6167

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature

## What is the current behavior?

- 

## What is the new behavior?

- `TabView` is up to date with WinUI master
- `IconSource` and its descendants are now implemented under WUX as well as MUX

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabViewItem.cs [hunk 1]
```diff
@@ -135,6 +154,8 @@ private void OnIsSelectedPropertyChanged(DependencyObject sender, DependencyProp
 		private void UpdateShadow()
 		{
 			if (SharedHelpers.IsThemeShadowAvailable())
+			// TODO Uno: Can't access XamlControlsResources from Uno.UI
+			//&& !Microsoft.UI.Xaml.Controls.XamlControlsResources.IsUsingControlsResourcesVersion2)
```
Comment (jeromelaban) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabViewItem.cs:158]: Once linking is available, we'll probably be able to merge fluent styles back to Uno.UI

**Review by jeromelaban — APPROVED** [review]

**Review by davidjohnoliver — APPROVED** [review]

**Comment by carldebilly** [comment]
@Mergifyio backport release/stable/3.7

**Comment by carlh98** [comment]
@carldebilly @jeromelaban Could we try backporting to 3.8?

**Comment by jeromelaban** [comment]
@Mergifyio backport release/stable/3.8

**Pull request merged**